### DISCUSSION TITLE
ONVIF Profile S

### DISCUSSION THREAD

**Discussion by Bob-AOL** [gtxaspec/wz_mini_hacks#660] [Development]
Is there still consideration being given to even a partial ONVIF implementation into wz_mini for the v3s?

I ask because I use eight v3s spread across 3 DVR/NVRs and Blue Iris with varying degrees (little to none) of useful camera functionality or even configuration parameters) being recognized by the hardware -- let alone manageable for additional feature optimization relative to the platform and particular application.

Many NVRs would be very happy to actually pull v3 configurations and become friends. I know mine certainly would.

**Comment by virmaior** [comment]
I just looked around a bit ... since basically a big part of this is producing the right XML.
Link to the spec: https://www.onvif.org/wp-content/uploads/2019/12/ONVIF_Profile_-S_Specification_v1-3.pdf?441d4a&441d4a

That being said, the ONVIF standard seems to be very over-documented and at the same time under-specified....

Is this the sort of package we're looking for https://github.com/mpromonet/v4l2onvif ?

**Reply by Bob-AOL** [reply]
> I just looked around a bit ... since basically a big part of this is producing the right XML. Link to the spec: https://www.onvif.org/wp-content/uploads/2019/12/ONVIF_Profile_-S_Specification_v1-3.pdf?441d4a&441d4a

I'm familiar with it. By contrast, Pelco offers a fairly cohesive real world overview.
 
> That being said, the ONVIF standard seems to be very over-documented and at the same time under-specified....

...,.and implemented in more and more IP streaming devices daily including the cheapie TP-Link Tapo cams, though ONVIF is listed as a protocol instead of a standard. The documentation is broad due to the inherent nature of the ongoing development. The 'standard' is specific on one level and optionally flexible on another to allow implementation of various complexities to be offered for various device applications and requirements -- ONVIF 'feature sets' as it were. The Pelco piece covers that fairly well.
 
> Is this the sort of package we're looking for https://github.com/mpromonet/v4l2onvif ?

I would think a more recent and readily available approach would be accessible somewhere these days. Perhaps a Tapo-C100 hack would be somewhere to start? I'm certainly not the one to ask. LOL

More control and interoperability between all these hot rodded wz_mini V3s out there and all the cool new ONVIF NVRs and VMS platforms sure would be cool, though.

Considering what you developers have already accomplished with the Wyze platform, it seems like adding Profile S ONVIF features would be trivial by comparison.

**Reply by Bob-AOL** [reply]
So has anyone taken a good look at the TP-Link Tapo C110 for ideas on possibly implementing some basic ONVIF features into wz_mini? We're talking a 20 dollar 2K camera with no need for internet access after initial registration that offers RTSP/ONVIF straight out of the box and drops right into Blue Iris and any other ONVIF capable VMS platforms including my three Amcrest XVR units. Quite amazing, actually.

I bought one just to mess with as a flu temperature monitor for my wood burning furnace and it holds its own side-by-side with a v.3 -- but with default RTSP/ONVIF capabilities. Pretty cool little cam. 

https://www.amazon.com/dp/B09YL5G1Y8?ref=ppx_yo2ov_dt_b_product_details&th=1

Comments? @gtxaspec or anyone else? Wz_mini ONVIF would could be a compatibility and control game changer for everyone.

**Reply by gtxaspec** [reply]
I'm delving into experiments with ONVIF, albeit on OpenIPC not wyze, but the same hardware.  Maybe we can get something working with wyze.  https://github.com/roleoroleo/onvif_simple_server

**Reply by Bob-AOL** [reply]
Cool. Here's hoping the endeavor is both enlightening and productive. It would be sooo nice for my XVRs to _not_  just arbitrarily think my v.3s are QVGA devices!
